contracts/src/PBHVerifier.sol

```diff
@@ -1,32 +1,176 @@
-    ///        the proxy in a way similar to a traditional solidity constructor.
-    constructor(address _logic, bytes memory _data) payable WorldIDProxy(_logic, _data) {
-        // !!!! DO NOT PUT PROGRAM LOGIC HERE !!!!
-        // It should go in the `initialize` function of the delegate instead.
+    /// @param root The root of the Merkle tree that this proof is valid for.
+    /// @param sender The sender of this particular transaction or UserOp.
+    /// @param nonce Transaction/UserOp nonce.
+    /// @param callData Transaction/UserOp call data.
+    /// @param pbhExternalNullifier External nullifier encoding month, year, and a pbhNonce.
+    /// @param nullifierHash Nullifier hash for this semaphore proof.
+    /// @param proof The zero-knowledge proof that demonstrates the claimer is registered with World ID.
+    event PBH(
+        uint256 indexed root,
+        address indexed sender,
+        uint256 nonce,
+        bytes callData,
+        uint256 indexed pbhExternalNullifier,
+        uint256 nullifierHash,
+        uint256[8] proof
+    );
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Vars                                  ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @dev Gap for transient storage.
+    uint256 private __gap;
+
+    /// @dev The World ID group ID (always 1)
+    uint256 internal immutable _GROUP_ID = 1;
+
+    /// @dev The World ID instance that will be used for verifying proofs
+    IWorldIDGroups internal _worldId;
+
+    /// @dev The EntryPoint where Aggregated PBH Bundles will be proxied to.
+    IEntryPoint internal _entryPoint;
+
+    /// @notice The number of PBH transactions that may be used by a single
+    ///         World ID in a given month.
+    uint8 public numPbhPerMonth;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Mappings                              ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @dev Whether a nullifier hash has been used already. Used to guarantee an action is only performed once by a single person
+    mapping(uint256 => bool) public nullifierHashes;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Functions                             ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @param sender The sender of this particular transaction or UserOp.
+    /// @param nonce Transaction/UserOp nonce.
+    /// @param callData Transaction/UserOp call data.
+    /// @param pbhPayload The PBH payload containing the proof data.
+    function verifyPbhProof(address sender, uint256 nonce, bytes memory callData, PBHPayload memory pbhPayload)
```

Comment:
nit: can we set `pbhPayload` as `calldata` instead of memory? Also maybe we update this function to `verifyPbh` for clarity as its not always verifying the proof.

Reply:
I think so

Reply:
It cannot be calldata